code/modules/clothing/spacesuits/flightsuit.dm

```diff
@@ -0,0 +1,796 @@
+	else
+		suit.slowdown = slowdown_air
+
+/obj/item/device/flightpack/process()
+	check_conditions()
+	handle_flight()
+	update_slowdown()
+	calculate_momentum_speed()
+	momentum_drift_tick++
+	momentum_drift()
+	handle_boost()
+	handle_damage()
+	update_icon()
+
+/obj/item/device/flightpack/proc/handle_damage()
+	if(crash_damage)
+		crash_damage = Clamp(crash_damage-crash_heal_amount, 0, crash_disable_threshold*10)
+		if(crash_damage >= crash_disable_threshold)
+			crash_disabled = 1
+		if(crash_disabled && (crash_damage <= 1))
+			crash_disabled = 0
+			crash_disable_message = 0
+			wearer << "<span class='notice'>Flightpack: Stabilizers re-calibrated. Flightpack re-enabled.</span>"
+	if(emp_damage)
+		emp_damage = Clamp(emp_damage-emp_heal_amount, 0, emp_disable_threshold * 10)
+		if(emp_damage >= emp_disable_threshold)
+			emp_disabled = 1
+		if(emp_disabled && (emp_damage <= 0.5))
+			emp_disabled = 0
+			wearer << "<span class='notice'>Flightpack: Systems rebooted. Flightpack re-enabled.</span>"
+	disabled = crash_disabled + emp_disabled
+	if(disabled)
+		if(crash_disabled && (!crash_disable_message))
+			wearer << "<span class='userdanger'>Flightpack: WARNING: STABILIZERS DAMAGED. UNABLE TO CONTINUE OPERATION. PLEASE WAIT FOR AUTOMATIC RECALIBRATION.</span>"
+			wearer << "<span class='warning'>Your flightpack abruptly shuts off!</span>"
+			crash_disable_message = 1
+		if(emp_disabled)
+			wearer << "<span class='userdanger'>Flightpack: WARNING: POWER SURGE DETECTED FROM INTERNAL SHORT CIRCUIT. PLEASE WAIT FOR AUTOMATIC REBOOT.</span>"
+			wearer << "<span class='warning'>Your flightpack abruptly shuts off!</span>"
+			emp_disable_message = 1
+		if(flight)
+			disable_flight(1)
+
+/obj/item/device/flightpack/update_icon()
+	if(!flight)
+		icon_state = initial(icon_state)
+		item_state = initial(item_state)
+	if(flight)
+		icon_state = icon_state_active
+		item_state = item_state_active
+		if(boost)
+			icon_state = icon_state_boost
+			item_state = item_state_boost
+	wearer.update_inv_wear_suit()
+	..()
+
+/obj/item/device/flightpack/proc/handle_flight()
+	if(!flight)
+		return 0
+	wearer.float(2)
+
+/obj/item/device/flightpack/proc/handle_boost()
+	if(boost)
+		boost_charge = Clamp(boost_charge-boost_drain, 0, boost_maxcharge)
+		if(boost_charge < 1)
+			deactivate_booster()
+	if(boost_charge < boost_maxcharge)
+		boost_charge = Clamp(boost_charge+boost_chargerate, 0, boost_maxcharge)
+
+
+/obj/item/device/flightpack/proc/cycle_power()
+	if(powersetting < powersetting_high)
+		powersetting++
+	else
+		powersetting = 1
+	momentum_gain = powersetting * 10
+	if(suit)
+		if(suit.user)
+			wearer << "<span class='notice'>FLIGHTPACK: Engines set to force [momentum_gain].</span>"
+
+
+/obj/item/device/flightpack/proc/losecontrol()
+	wearer.visible_message("<span class='warning'>[wearer]'s flight suit careens wildly as they lose control of it!</span>")
+	if(wearer)
+		while(momentum_x != 0 || momentum_y != 0)
+			spawn(2)
```

Comment:
this is probably going to be replaced by some form of throwing as this is not nearly lethal enough.